pkg/gather/clusterconfig/csr.go

```diff
@@ -0,0 +1,185 @@
+package clusterconfig
+
+import (
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"time"
+
+	"github.com/openshift/insights-operator/pkg/utils"
+	"k8s.io/api/certificates/v1beta1"
+	certificatesv1b1api "k8s.io/api/certificates/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/json"
+)
+
+type CSRAnonymizer struct {
+	*certificatesv1b1api.CertificateSigningRequest
+}
+
+func (a CSRAnonymizer) Marshal(_ context.Context) ([]byte, error) {
+	res, err := anonymizeCsr(a.CertificateSigningRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	return json.Marshal(res)
+}
+
+func anonymizeCsrRequest(r *certificatesv1b1api.CertificateSigningRequest, c *CSRAnonymizedFeatures) {
+	if r == nil || c == nil {
+		return
+	}
+	c.Spec = &StateFeatures{}
+	c.Spec.Username = r.Spec.Username
+	c.Spec.Groups = r.Spec.Groups
+	c.Spec.Usages = r.Spec.Usages
+
+	// CSR in a PEM
+	// parse only first PEM block
+	block, _ := pem.Decode(r.Spec.Request)
+	if block == nil {
+		// unable to decode CSR: missing block
+		return
+	}
+	csr, err := x509.ParseCertificateRequest(block.Bytes)
+	if err != nil {
+		return
+	}
+
+	err = csr.CheckSignature()
+	if err != nil {
+		return
+	}
+	c.Spec.Request = &CsrFeatures{}
+	c.Spec.Request.ValidSignature = err == nil
+	c.Spec.Request.Subject = anonymizePkxName(csr.Subject)
+
+	c.Spec.Request.SignatureAlgorithm = csr.SignatureAlgorithm.String()
+	c.Spec.Request.PublicKeyAlgorithm = csr.PublicKeyAlgorithm.String()
+	c.Spec.Request.DNSNames = utils.Map(csr.DNSNames, anonymizeURL)
+	c.Spec.Request.EmailAddresses = utils.Map(csr.EmailAddresses, anonymizeURL)
+	ipsl := []string{}
```

Comment:
if we know `len(csr.IPAddresses)`, you can allocate an array, but it's up to you